# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTypes.h
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLTypes.h
--- a/dom/canvas/WebGLTypes.h	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLTypes.h	Wed Jan 08 22:19:23 2020 +0000
@@ -20,6 +20,7 @@
 #include "gfxTypes.h"
 
 #include "nsTArray.h"
+#include "nsString.h"
 #include "mozilla/dom/WebGLRenderingContextBinding.h"
 #include "mozilla/ipc/SharedMemoryBasic.h"
 //#include "WebGLStrongTypes.h"
@@ -515,6 +516,7 @@
   bool resistFingerprinting = false;
   uvec2 size = {};
   WebGLContextOptions options;
+  uint32_t principalKey = 0;
 };
 
 struct Limits final {
@@ -887,14 +889,6 @@
   return {abv.DataAllowShared(), abv.LengthAllowShared()};
 }
 
-template <typename T>
-inline Range<const uint8_t> MakeByteRange(const T& x) {
-  const auto typed = MakeRange(x);
-  return Range<const uint8_t>(
-      reinterpret_cast<const uint8_t*>(typed.begin().get()),
-      typed.length() * sizeof(typed[0]));
-}
-
 Maybe<Range<const uint8_t>> GetRangeFromView(const dom::ArrayBufferView& view,
                                              GLuint elemOffset,
                                              GLuint elemCountOverride);
@@ -927,6 +921,10 @@
 
 uint8_t ElemTypeComponents(GLenum elemType);
 
+inline std::string ToString(const nsACString& text) {
+  return {text.BeginReading(), text.Length()};
+}
+
 }  // namespace mozilla
 
 #endif